### ISSUE TITLE
globally unique names

### ISSUE THREAD

**Issue by progrium** [docker/notary#22]
This particular approach to this pattern is very familiar but I'm not sure if you're making it up or what the proper semantics are. What happens if you go to a GUN over HTTP? Should you prefer making GUNs that map to URLs? Is there are friendlier name for this idea?

**Comment by endophage** [comment]
The names are really "Globally Unique" to the specific notary-server being published to. The plan is to have local configuration that allows 2 users with identical GUNs in their local store, to point them at different notary-servers, potentially receiving different information.

To support this, GUNs would become an entirely local concept and would locally be mapped to a unique ID (maybe a public key?), which would then be used to query a notary-server.

Very interested to hear your thoughts about that idea!

**Comment by progrium** [comment]
So I'm guessing they will eventually not be called "GUNs" since they are local to notary-servers? What is the plan for notary server infrastructure? I'm guessing anybody can run their own? Maybe the domain part of the GUN as it is now would map to a notary server?

PS, I've been working on a way to have unique local names made global under a namespace that identifies you ... for magic discovery and rendezvous. Might be something useful to this down the road.

**Comment by endophage** [comment]
GUN's will not necessarily have a domain part. The GUN is used to create a local directory structure, which we intend to populate with configuration among other things. The heirarchical configuration mentioned in issue #1 will allow a user to point any sub-tree of a GUN to any notary server.

It's a very academic topic, but really nailing down what "global" means is difficult, especially when we want these names to be human readable (and therefore sufficiently memorable). The only concrete requirement for "global" is that when the same user types in the same name multiple times, it references the same thing, unless that user has explicitly changed their configuration. I'm definitely interested in your ideas about making local name available under an identifiable namespace. 

I just noticed your question about friendliness of the name. We originally used FQDN (Fully Qualified Docker Name) but this was obviously confusing, we tried shortening it to QDN, that wasn't much better. We're open to other names, we came up with "Globally Unique Name" as it described the concept we're aiming for without specifically referencing "Docker", as we don't want people to think Notary is a tool only for container signing (a question I've had quite a few times since DockerCon).
